types/milesight/WS202/uplink.js

```diff
@@ -0,0 +1,42 @@
+// Parse Hex Byte Array
+function parseHexString(str) {
+  const result = [];
+  while (str.length >= 2) {
+    result.push(parseInt(str.substring(0, 2), 16));
+    str = str.substring(2, str.length);
+  }
+  return result;
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const bytes = parseHexString(payload);
+
+  const data = {};
+  const lifecycle = {};
+
+  for (let i = 0; i < bytes.length; ) {
+    const channelId = bytes[i++];
+    const channelType = bytes[i++];
+    // BATTERY
+    if (channelId === 0x01 && channelType === 0x75) {
+      lifecycle.batteryLevel = bytes[i];
+      i += 1;
+    }
+    // PIR
+    else if (channelId === 0x03 && channelType === 0x00) {
+      data.pir = Number(bytes[i]);
```

Comment:
not sure this is gonna work as intended :thinking:

Reply:
Seems to work, would you suggest to do this without type casting ?

Reply:
The bytes array contains f.e the following values:  [
  1,
  117,
  16,
  3,
  0,
  1,
  4,
  0,
  0,
]
It's one byte per entry but not encoded as binary.

Reply:
ok, so the values are already parsed from hex to numbers? make sure they are numbers and use parseInt() (or double or whatever) if you want to convert something to numbers ^^

Reply:
👍 they get parsed automatically but the decoder expects this.